src/lighting/TrainLighting.cpp

```diff
@@ -0,0 +1,307 @@
+    }
+
+    void TrainLighting::_do_update_internal_mover(TMoverParameters *mover) {
+        ASSERT_MOVER(mover);
+        TrainPart::_do_update_internal_mover(mover);
+        mover->LightsPosNo = static_cast<int>(light_position_list.size());
+        mover->LightsWrap = wrap_light_selector;
+        mover->LightsDefPos = default_selector_position;
+        mover->LightPower = 0; //LightPower is used there but declared in the Param section in the .fiz file
+        mover->LightPowerSource.SourceType = static_cast<TPowerSource>(light_source);
+        mover->AlterLightPowerSource.SourceType = static_cast<TPowerSource>(alternative_light_source);
+        mover->LightsPos = selector_position;
+        // Enable or disable lighting whether required power is available or not
+        switch (light_source) {// I don't know as for now if this is going to eb used elsewhere, but it was a part of the Mover
+            case TrainController::POWER_SOURCE_INTERNAL:
+            case TrainController::POWER_SOURCE_TRANSDUCER:
+            case TrainController::POWER_SOURCE_GENERATOR:
+            case TrainController::POWER_SOURCE_POWERCABLE:
+            case TrainController::POWER_SOURCE_ACCUMULATOR:
+            case TrainController::POWER_SOURCE_HEATER:
+            case TrainController::POWER_SOURCE_CURRENTCOLLECTOR:
+            case TrainController::POWER_SOURCE_NOT_DEFINED:
+                _is_powered = mover->Power24vIsAvailable || mover->Power110vIsAvailable;
+                break;
+            case TrainController::POWER_SOURCE_MAIN:
+                _is_powered = mover->Mains;
+                break;
+            default:;
+        }
+        //FIXME: HeadLight section is supported there but https://github.com/eu07/maszyna/blob/9aad41c0ba5a200a2935b578201dfc2f10d70662/McZapkie/Mover.cpp#L10044C31-L10044C33 original mover is probably not using it anywhere since it's trying to extract config values into non-existing variables :)
+        //FIXME: Might be incomplete, needs further testing. Taking a look into MaSzyna's Train.cpp might be helpful
+        //@TODO: Handle any power changes using TrainController
+    }
+
+    void TrainLighting::_do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state) {
+        ASSERT_MOVER(mover);
+        state["light_position"] = mover->LightsPosNo;
+        state["light_power"] = mover->LightPower;
+        state["power_source"] = static_cast<TrainController::TrainPowerSource>(mover->LightPowerSource.SourceType);
+    }
+
+    void TrainLighting::_do_fetch_config_from_mover(TMoverParameters *mover, Dictionary &config) {
+        TrainPart::_do_fetch_config_from_mover(mover, config);
+    }
+
+    void TrainLighting::_register_commands() {
+        register_command("set_light_selector_position", Callable(this, "set_light_selector_position"));
+        register_command("get_light_selector_position", Callable(this, "set_light_selector_position"));
```

Comment:
Commands are mutators only. There are no separate commands for set and get. The `light_selector_position` should be returned in the state only.